CppCoreGuidelines.md

```diff
@@ -14084,7 +14084,7 @@ this can be a security risk.
 
 Some is possible, do at least something.
 There are commercial and open-source tools that try to address this problem,
-but be aware that solutions have costs and blind spots.
```

Comment:
While the phrase `blind spot` uses the word `blind`, I don't believe that it is an offensive term, as all humans have a [blind spot](https://en.wikipedia.org/wiki/Blind_spot_(vision)) in each eye. Blind spots also occur elsewhere, such as in [driving](https://en.wikipedia.org/wiki/Vehicle_blind_spot), therefore while this phrase does contain the word `blind`, I think it conveys a different message.

Reply:
In my personal opinion: I am not sure. In online articles, opinions go both ways, and that alone seems enough for me to err on the side of caution in this case. The way I understand the side of the argument against usages such as here is that the word "blind" here is connected to a negative meaning, e.g. marked by "but be aware", and the insinuation that this weakness cannot be compensated. I understand that the combined expression "blind spot" does not derive from blindness as a disability, but etymology is often regarded as outweighed by psychology of language when it comes to inclusive language. In other words, not all recommendations for inclusive language follow strict word-meaning logic, but rather the impact of words on emotions and subconscious biases. As an example "blacklist" has no racial meaning in etymology, but is often considered non-inclusive because of the multiple meanings of "black". But that's only my personal interpretation, there is no hard data or authority. Like with other metaphors, erring on the side of caution here probably has low negative impact. If the text after the change is at least as clear as before, I am not sure what is to be gained by keeping the original version.

Reply:
I agree with @Lirusaito's argument but I think regardless the new phrasing is probably better.

I do think it's weird that this sentence identifies costs and false negatives as problems and then next sentence seems to elaborate by talking about costs and false _positives_. Perhaps reworking these two sentences in combination would yield a more palatable solution. For example

> There are commercial and open-source tools that try to address this problem, but static tools often have many false positives and run-time tools often have a significant cost.

Reply:
> and that alone seems enough for me to err on the side of caution in this case. 

I usually try to stay away from such arguments because I'm neither living in the US, nor am I an English native speaker, so there are certainly things I don't quite understand and I'm probably biased and/or missing cultural context. 

However, I'd really like to point out: Just because someone (or a group of someones) claims there is a problem isn't enough evidence that there actually is a problem and it is imho no a good idea, to base such decisions (to use or not to use a particular word) on Fear, Uncertainty and Doubt that something might, potentially,  somewhere, maybe have a problematic effect on someone. I certainly don't want to - in any way - suggest that the arguments for inclusive language in general are FUD. I'm just saying, that if - in this particular case - there isn't any clear evidence that a concrete term is problematic (in other words, if the jury is still out) and it is otherwise the best way to convey the desired meaning, it should not be changed just out of fear it could be problematic. Very much the same way you should not start to spend hours and hours to optimize the performance of your code at the cost of readability without clear evidence that it was slow to begin with.

I completely agree, that derogative terms have no place in technical writing, but just outright avoiding any terms, idioms and words that can be used to describe a person is imho swinging the pendulum far too much into the opposite direction. 

TL;DR: My point is: Imo "Blind spot" is a perfectly legal technical term and we should not be afraid to use it when it is appropriate. 

Of course that doesn't preclude that - irrespective of the issue of inclusive language - there might be a better way to convey the meaning of that bullet point and I have no problem with the particular change here. It should just not happen only based on the vague possibility that we might do something wrong.

Reply:
I believe "blind spot" is usually used ambiguously. It has accumulated too many slightly different meanings in history. Even in biology, it can both mean the area of the visual field, or an area of the retina. Where it has a precisely defined meaning (such as in driving or in radio transmission), it can be used like "The so-called 'blind spot' in driving is...". But "so-called blind spot" does not work for software analysers, there is no such definition. It's not a clear technical term in software, it's meaning cannot be looked up. For figurative use, I found multiple broad definitions such as "An area in which a person lacks understanding or impartiality." and "an area in which one fails to exercise judgment or discrimination".

> if there isn't any clear evidence that a concrete term is problematic ...

Clear evidence is great for decision-making. Sadly though, behavioural causes for continued discrimination are very diffuse and difficult to prove in sociology and psychology. Waiting for clear evidence before action does not benefit those who are discriminated against. Because "blind spot" in the figurative sense does not seem to ever be clarifying to me (explained above), I would not be sad to see it generally avoided (outside biological texts), even if I never get hard evidence that it is problematic.

Reply:
Fair enough.Probably have to spend some more thought on that topic

Reply:
Editors think "blind spot" is a neutral technical jargon term (and medical and automotive).

Reply:
reverted